### PR TITLE
feat(TDP-2049): Edit a Database dataset properties

### PR DESCRIPTION
**Pull request by frassinier**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-2049

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [x] The new code does not introduce new technical issues (sonar / eslint)
- [ ] Functional tests have been performed

**Please check the browsers you've tested on**
- [x] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

**(Optional) What is the current behavior?**
(Additional information to the Jira)

**(Optional) What is the new behavior?**
(Additional information to the Jira)

**(Optional) Other information**:
Will fix 
* https://jira.talendforge.org/browse/TDP-3132
* https://jira.talendforge.org/browse/TDP-3109

### CODE REVIEW

**Comment by build-talend** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/64/) [:x:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/64/)

**Comment by build-talend** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/66/) [:x:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/66/)

**Comment by build-talend** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/100/) [:x:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/100/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/189/) [:x:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/189/)

File: dataprep-webapp/src/app/settings/actions/dataset-actions-service.js [hunk 1]
```diff
@@ -125,8 +125,23 @@ export default class DatasetActionsService {
...
+			this.StateService.setDatasetToUpdate(payload.model);
+			break;
+		}
+		case '@@dataset/TCOMP_EDIT': {
+			const { payload } = action;
+			if (payload.model) {
+				this.StateService.setCurrentImportItem(payload.model);
+				this.StateService.showImport();
+			}
+			else {
```
Comment (jsomsanith) [inline dataprep-webapp/src/app/settings/actions/dataset-actions-service.js]: When don't we have a model ?

File: dataprep-webapp/src/app/services/import/import-service.js [hunk 2]
```diff
@@ -15,38 +15,25 @@
...
 export default class ImportService {
 
-	constructor($document, $rootScope, $translate, state, DatasetService, ImportRestService,
-				StateService, TalendConfirmService, UploadWorkflowService, UpdateWorkflowService) {
+	constructor($document, $rootScope, $translate, state, DatasetService, ImportRestService, StateService, TalendConfirmService, UploadWorkflowService, UpdateWorkflowService) {
 		'ngInject';
 
 		this.$document = $document;
 		this.$rootScope = $rootScope;
 		this.$translate = $translate;
```
Comment (jsomsanith) [inline dataprep-webapp/src/app/services/import/import-service.js]: $translate is not used anymore

File: dataprep-webapp/src/app/components/home/dataset/react-home-dataset.html [hunk 3]
```diff
@@ -48,11 +48,18 @@
         state="$ctrl.state.import.visible"
         ng-if="$ctrl.state.import.visible"
         close-button="true"
+        on-close="$ctrl.ImportService.StateService.setCurrentImportItem(null)"
```
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/home/dataset/react-home-dataset.html]: can you create a $ctrl.resetImport() that call directly StateService ?
This becomes unclear here

File: dataprep-webapp/src/app/services/state/import/import-state-service.js [hunk 4]
```diff
@@ -8,12 +8,14 @@
  ============================================================================*/
 
 export const importState = {
+	importItem: null,
```
Comment (jsomsanith) [inline dataprep-webapp/src/app/services/state/import/import-state-service.js]: Rename this to `item` please (unnecessary context)

File: dataprep-webapp/src/app/services/import/import-rest-service.js [hunk 5]
```diff
@@ -35,18 +36,29 @@ export default function ImportRestService($http, RestURLs) {
...
+	}
+
+	/**
+	 * @ngdoc method
+	 * @name refreshForm
+	 * @methodOf data-prep.services.import.service:ImportRestService
+	 * @description Refresh the form
+	 * @returns {Promise}  The POST call promise
+	 */
+	function refreshForm(definitionName, propertyName, formData) {
```
Comment (jsomsanith) [inline dataprep-webapp/src/app/services/import/import-rest-service.js]: `definitionName` is not used

File: dataprep-webapp/src/app/settings/actions/dataset-actions-service.js (hunk 1 above)
Comment (frassinier) [inline dataprep-webapp/src/app/settings/actions/dataset-actions-service.js]: Indeed, never.

File: dataprep-webapp/src/app/components/dataset/import/tcomp/dataset-import-tcomp.controller.spec.js [hunk 6]
```diff
@@ -0,0 +1,293 @@
...
+ This source code is available under agreement available at
+ https://github.com/Talend/data-prep/blob/master/LICENSE
+
+ You should have received a copy of the agreement
+ along with this program; if not, write to Talend SA
+ 9 rue Pages 92150 Suresnes, France
+
+ ============================================================================*/
+
+describe('Dataset Import TCOMP controller', () => {
```
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/dataset/import/tcomp/dataset-import-tcomp.controller.spec.js]: Missing tests on `$onChanges`

**Pull request merged**